presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/PropertyCacher.kt

```diff
@@ -0,0 +1,37 @@
+package com.exallium.mvvmcleanexample.presentation.utils
+
+import kotlin.reflect.KProperty
+
+object PropertyCacher {
```

Comment:
Javadoc.

What does this do for us? Why is it necessary?

To someone coming into the codebase, this reads like a bunch of boilerplate.